folium/features.py

```diff
@@ -538,7 +538,7 @@ def process_data(self, data):
         elif hasattr(data, '__geo_interface__'):
             self.embed = True
             if hasattr(data, 'to_crs'):
-                data = data.to_crs('EPSG:4326')
+                data.crs = 'EPSG:4326'
```

Comment:
This changes the behaviour of the function. `folium` requires geometry in 4326 which is ensured by the `to_crs` call. This change overrides the CRS without reprojection.

The existing code seems to be perfectly fine and in line with the current geopandas API.

The only case where this may cause trouble is a GeoDataFrame with `None` as a CRS. In such a case `to_crs` will raise a warning and could be skipped here (assuming the geometries are in the correct CRS).

Reply:
> The only case where this may cause trouble is a GeoDataFrame with None as a CRS. In such a case to_crs will raise a warning and could be skipped here (assuming the geometries are in the correct CRS).

IMO that should fail and the user should add a crs explicitly to avoid implicit assumptions.

Reply:
Thanks so much for the clarification! I will close this then